internal/core/src/storage/Event.cpp

```diff
@@ -282,6 +283,17 @@ BaseEventData::Serialize() {
             }
             break;
         }
+        case DataType::GEOSPATIAL: {
+            for (size_t offset = 0; offset < field_data->get_num_rows();
+                 ++offset) {
+                auto geo_ptr = static_cast<const std::string*>(
+                    field_data->RawValue(offset));
+                payload_writer->add_one_binary_payload(
```

Comment:
change it to the size to" auto size =
                    field_data->is_valid(offset) ? string_view.size() : -1;", which can help to add null in payload.